hack/deploy.sh

```diff
@@ -5,71 +5,46 @@ set -e
-        echo "Openshift CI detected, deploying using image $FULL_REGISTRY_IMAGE"                   
-fi   
+# expect kustomize to be in PATH by default
+KUSTOMIZE="${KUSTOMIZE:-kustomize}"
+
+# Override the image name when this is invoked from openshift ci
+if [ -n "${OPENSHIFT_BUILD_NAMESPACE}" ]; then
+  FULL_REGISTRY_IMAGE="registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:performance-addon-operators-registry"
+fi
+
+echo "Deploying using image $FULL_REGISTRY_IMAGE."
+
+# Deploy features
+success=0
+iterations=0
+sleep_time=10
+max_iterations=30 # results in 5 minute timeout
+feature_dir=cluster-setup/ci-cluster/performance/
+
+until [[ $success -eq 1 ]] || [[ $iterations -eq $max_iterations ]]
+do
+
+  echo "[INFO] Deploying performance operator and profile."
+  set +e
+  if ! ${KUSTOMIZE} build $feature_dir | envsubst | ${OC_TOOL} apply -f -
+  then
+    set -e
+    iterations=$((iterations + 1))
+    iterations_left=$((max_iterations - iterations))
+    echo "[WARN] Deployment failed, retrying in $sleep_time sec, $iterations_left retries left."
+    sleep $sleep_time
+    continue
+  fi
+  set -e
+
+  success=1
 
-${OC_TOOL} apply -f - <<EOF
----
-apiVersion: v1
-kind: Namespace
-metadata:
-  labels:
-    openshift.io/cluster-monitoring: "true"
-  name: openshift-performance-addon
-spec: {}
-EOF
-
-${OC_TOOL} apply -f - <<EOF
----
-apiVersion: operators.coreos.com/v1
-kind: OperatorGroup
-metadata:
-  name: openshift-performance-addon-operatorgroup
-  namespace: openshift-performance-addon
-spec:
-  targetNamespaces:
-  - openshift-performance-addon
-EOF
-  
-${OC_TOOL} apply -f - <<EOF
----
-apiVersion: operators.coreos.com/v1alpha1
-kind: CatalogSource
-metadata:
-  name: performance-addon-operators-catalogsource
-  namespace: openshift-marketplace
-spec:
-  displayName: Openshift Performance Addon Operators
-  icon:
-    base64data: ""
-    mediatype: ""
-  image: ${FULL_REGISTRY_IMAGE}
-  publisher: Red Hat
-  sourceType: grpc
-EOF
-
-${OC_TOOL} apply -f - <<EOF
-apiVersion: operators.coreos.com/v1alpha1
-kind: Subscription
-metadata:
-  name: performance-addon-operators-subscription
-  namespace: openshift-performance-addon
-spec:
-  channel: alpha
-  name: performance-addon-operators
-  source: performance-addon-operators-catalogsource
-  sourceNamespace: openshift-marketplace
-EOF
-
-# Wait for performance-addon-operator deployment to be ready
-until ${OC_TOOL} -n openshift-performance-addon get deploy/performance-operator; do
-    echo "[INFO]: get performance-operator deployment"
-    sleep 10
 done
-${OC_TOOL} -n openshift-performance-addon wait deploy/performance-operator --for condition=Available --timeout 5m
 
-# Deploy performance-profile custom resource
-${KUSTOMIZE} build cluster-setup/ci-cluster/performance/ | envsubst | ${OC_TOOL} apply -f -
+if [[ $success -eq 0 ]]; then
+  echo "[ERROR] Deployment failed, giving up."
+  exit 1
+fi
+
```

Comment:
Do we want to handle the worker labeling and waiting on MCP to settle down in this deploy script as well?  that seems logical considering we're posting the performance profile in this step along with the operator.

Reply:
I have labeling and waiting in 2 different scripts now and made the functest target depend on them... good enough?